GitExtUtils/GitUI/UIExtensions.cs

```diff
@@ -47,5 +50,34 @@ public static bool IsFixedWidth(this Font ft, Graphics g)
 
             return fixedWidth;
         }
+
+        public static IEnumerable<ListViewItem> SelectedItems(this ListView listView) =>
+            listView.SelectedItems.Cast<ListViewItem>();
+
+        public static IEnumerable<ListViewItem> Items(this ListView listView) =>
+            listView.Items.Cast<ListViewItem>();
+
+        public static IEnumerable<ListViewGroup> Groups(this ListView listView) =>
+            listView.Groups.Cast<ListViewGroup>();
+
+        public static T Tag<T>(this ListViewItem item) =>
+            (T)item.Tag;
+
+        public static T Tag<T>(this ListViewGroup grp) =>
+            (T)grp.Tag;
+
+        public static IEnumerable<T> ItemTags<T>(this ListView listView) =>
+            listView.Items().Select(Tag<T>);
+
+        public static IEnumerable<T> SelectedItemTags<T>(this ListView listView) =>
+            listView.SelectedItems().Select(Tag<T>);
+
+        /// <summary>
+        /// <para>For practical purposes: The last <see cref="ListViewItem"/> added to selection.</para>
+        /// <para>Actually: Focused item if selected, otherwise first selected item</para>
+        /// </summary>
+        public static ListViewItem LastSelectedItem(this ListView listView) =>
+            listView.FocusedItem?.If(_ => _.Selected) ??
+            listView.SelectedItems().FirstOrDefault();
```

Comment:
Let's move it out to `ListViewExtensions` or something like that

Reply:
ok